### PR TITLE
feat: esm-only

### LINKED ISSUE #838: bug: Connector not found
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.6.2

### Current Behavior

I'm getting `Connector not found` error everytime I try to connect to metamask within the Metamask App browser on Android

### Expected Behavior

Expect connect dialog to show up in metamask

### Steps To Reproduce

- Open Metamask App on Android
- Visit the codesandbox and press metamask button

### Link to Minimal Reproducible Example (CodeSandbox, StackBlitz, etc.)

https://codesandbox.io/s/wagmi-test-forked-v5qi39

### Anything else?

_No response_

### LINKED ISSUE #1004: No wagmi client found in Code Sandbox
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.6.7

### Current Behavior

Hi! I wanted to create a minimally reproducible example in Code Sandbox for another issue, but encountered an error: `Error: No wagmi client found. Ensure you have set up a client: https://wagmi.sh/docs/client`. 

This happens if you try to connect MetaMask or WalletConnect

### Expected Behavior

Wagmi must connect wallet.

### Steps To Reproduce

1. Open Sandbox
2. Click on MetaMask or WalletConnect button.

### Link to Minimal Reproducible Example (CodeSandbox, StackBlitz, etc.)

https://codesandbox.io/s/wagmi-walletconnect-trustwallet-23wxfl?file=/src/App.tsx

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by tmm**
## Description

This PR converts `wagmi` from bundling both CJS + ESM, to bundling pure ESM. 

This will probably affect tooling that is sensitive to importing ESM such as Jest and Remix. We might need to create some additional guides on how to use wagmi with these tools as we move to pure ESM.

Evolution of #933 using [tsup](https://tsup.egoist.dev) instead of patching Preconstruct

Fixes #838 
Fixes #1004 

## Additional Information

- [x] I read the [contributing docs](/wagmi-dev/wagmi/blob/main/.github/CONTRIBUTING.md) (if this is your first contribution)

Your ENS/address: awkweb.eth

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -1,10 +0,0 @@
-module.exports = {
```
Comment (jxom) [inline babel.config.js:1]: this is exciting

File: docs/next.config.js [hunk 2]
```diff
@@ -39,12 +42,4 @@ const config = {
   },
 }
 
-if (process.env.NODE_ENV === 'development') {
```
Comment (jxom) [inline docs/next.config.js:42]: Will this still work with symlinks?

File: packages/react/tsup.config.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  'src/connectors/metaMask.ts',
+  'src/connectors/mock.ts',
+  'src/connectors/walletConnect.ts',
+  'src/providers/alchemy.ts',
+  'src/providers/public.ts',
+  'src/providers/infura.ts',
+  'src/providers/jsonRpc.ts',
+]
+
+export default defineConfig(() => {
```
Comment (jxom) [inline packages/react/tsup.config.ts:22]: Thoughts on moving this config up into a `tsup.config.ts` on the root, and then passing through the entries inside this file?

```tsx
import { getConfig } from '../../utils/tsup.ts'

const entry = [
  'src/index.ts',
  'src/actions.ts',
  'src/chains.ts',
  'src/connectors/coinbaseWallet.ts',
  'src/connectors/injected.ts',
  'src/connectors/metaMask.ts',
  'src/connectors/mock.ts',
  'src/connectors/walletConnect.ts',
  'src/providers/alchemy.ts',
  'src/providers/public.ts',
  'src/providers/infura.ts',
  'src/providers/jsonRpc.ts',
]

export default getConfig({ entry })
```

will be good if we have more packages down the line

File: packages/react/src/hooks/contracts/useContractInfiniteReads.ts [hunk 4]
```diff
@@ -95,7 +95,11 @@ export function useContractInfiniteReads<
   select,
   staleTime,
   suspense,
-}: UseContractInfiniteReadsConfig<TContracts, TPageParam>) {
+}: UseContractInfiniteReadsConfig<
```
Comment (jxom) [inline packages/react/src/hooks/contracts/useContractInfiniteReads.ts:98]: Are these changes from another PR?

File: docs/next.config.js (hunk 2 above)
Comment (tmm) [reply docs/next.config.js:42]: Can make it work! Only have the playground set up that way now, but can get the docs site going too.

File: packages/react/src/hooks/contracts/useContractInfiniteReads.ts (hunk 4 above)
Comment (tmm) [reply packages/react/src/hooks/contracts/useContractInfiniteReads.ts:98]: Yup! Let me pull in from main so it's not cluttering the diff here.

File: packages/react/tsup.config.ts (hunk 3 above)
Comment (tmm) [reply packages/react/tsup.config.ts]: good idea! added

**Review by jxom — APPROVED** [review]
Looks great! Would be sweet to have some docs around using wagmi w/ tooling that doesn't have first-class support for ESM such as Jest + Remix!

**Pull request merged**